src/core/thread/router_table.cpp

```diff
@@ -531,6 +531,7 @@ void RouterTable::ProcessTlv(const Mle::RouteTlv &aTlv)
             RemoveNeighbor(*router);
 
             mAllocatedRouterIds.Remove(routerId);
+            Get<AddressResolver>().Remove(routerId);
```

Comment:
`Get<AddressResolver>().Remove(routerId);` is already called by `RemoveNeighbor(*router);` ?

Reply:
The current code of handling `RouterTLV` and `ThreadRouterMaskTlv` are not exactly same. Should we just use one same code for processing both TLVs ?

Reply:
Agree that it is not necessary here and there are redundancies in router id set relative process. will close this one submit a separate one for enhancements.